platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,477 @@
+        the assets [from S3](https://github.com/department-of-veterans-affairs/vets-website/blob/006185b7c40fe85a8b219ae99f3aefb3d8bf0e09/src/site/stages/build/plugins/download-assets.js#L111)
+        - Process is triggered [from this Jenkins config](https://github.com/department-of-veterans-affairs/vets-website/blob/006185b7c40fe85a8b219ae99f3aefb3d8bf0e09/Jenkinsfile.content#L32)
+    - Does **NOT** run any content validation such as accessibility tests or
+      broken link checker
+2. [Full deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#full-deploy-of-vagov-client-app)
+    - For `vets-website` code changes
+    - Creates a new release and deploys it
+    - Fetches the latest Drupal content
+      - Falls back to the cached content in S3 from the last successful deploy
+        (full or partial)
+        <!-- TODO: Verify that either deploy will cache the pages.json and files -->
+    - Runs the accessibility tests and link checker
+
+See [the deployment documentation](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/)
+for more details.
+
+#### Pain points
+- Engineers encounter **confusing build errors** that are due to the mismatch in
+  outdated locally cached content and updated Liquid templates
+- Fetching content from the CMS and building HTML files when developing
+  applications **slows down the local builds** for engineers unnecessarily
+- It **requires either SOCKS proxy access** or the `npm run fetch-drupal-cache`
+  command, which isn't obvious
+- **Application deployments and PR builds fail** when content in the CMS is
+  invalid
+  - This includes broken links and accessibility violations
+  - We have a safeguard that mostly works; if we run into a failure using the
+    latest CMS content, we try again with the cached content in S3 from the last
+    successful build
+    - This doesn't work when the GraphQL query has been updated; that
+      invalidates the cache
+    - Errors in liquid templates would potentially break the build and falling
+      back to the cache won't help
+
+### High Level Design
+The current front end build will be split up into two distinct builds:
+1. Content build
+    - Input: CMS content
+    - Output: HTML files, assets from the CMS (PDFs, images and the like)
+    - This maps pretty cleanly to the current partial deploy
+1. Application build
+    - Input: `vets-website` code
+    - Output: JavaScript and CSS bundles
+    - This essentially maps to the current full deploy minus the content
+
+The output of both these builds will be deployed to a single S3 bucket with
+safeguards in place to ensure they don't override each other.
+
+**Another important note:** There is no way to coordinate these two deploys to
```

Comment:
I think we should work this paragraph some more. 

- use it to summarize the tradeoffs 
- tradeoffs: 
  - the two processes are no longer block each other when validations fail and no longer share data 
  - cache busting currently relies on using new file names in static content on every deploy. splitting the processes will require using a different cache busting technique like `cache-control` headers 
  - new application releases will require manual coordination between CMS and website deployment processes

Reply:
I see you have a cache busting section that can be referenced. Maybe just move this into the same level section as "new application release" and reference both sections here.

Reply:
I'm not really following on this one. What "new application release" section?